<launch>

  <arg name="world_frame_name" default="world" /> <!--world /turtlebot0/odom-->
  <arg name="map_name" default="/tentabot_map" />

  <!--turtlebot0
      firefly-->
  <arg name="map_frame_name" default="/turtlebot0/base_link" />
  <arg name="map_resolution" default="0.1" />
  <arg name="map_server_dt" default="0.2" />

  <!--/firefly/camera/depth/points
      /firefly/d435/camera/depth/color/points
      /turtlebot0/d435_cam1d435/camera/depth/color/points-->
  <arg name="sensor_pc2_msg_name" default="/turtlebot0/d435_cam1d435/camera/depth/color/points" />
  <arg name="sensor_pc2_direction" default="z" /> <!--x y z-->
  <arg name="sensor_pc2_min_range" default="0.2" />
  <arg name="sensor_pc2_max_range" default="30.0" />
  <arg name="sensor_pc2_max_yaw" default="0.8" />
  <arg name="sensor_pc2_max_pitch" default="1.2" />

  <!--/turtlebot0/scan 
      /firefly/scan-->
  <arg name="sensor_laser_msg_name" default="/turtlebot0/scan" />
  <arg name="sensor_laser_max_range" default="4.0" />

  <!--BASED ON THE ROBOT DIMENSIONS-->
  <!--turtlebot3-->
  <arg name="crop_x_max" default="0.15" />
  <arg name="crop_x_min" default="-0.15" />
  <arg name="crop_y_max" default="0.15" />
  <arg name="crop_y_min" default="-0.15" />
  <arg name="crop_z_max" default="0.5" />
  <arg name="crop_z_min" default="-0.0" />

  <!--firefly-->
  <!--arg name="crop_x_max" default="0.3" />
  <arg name="crop_x_min" default="-0.3" />
  <arg name="crop_y_max" default="0.3" />
  <arg name="crop_y_min" default="-0.3" />
  <arg name="crop_z_max" default="0.5" />
  <arg name="crop_z_min" default="-0.1" /-->

  <!--BASED ON THE OCCUPANCY GRID (ALSO SENSOR) AROUND THE ROBOT-->
  <arg name="bbx_x_max" default="10" />
  <arg name="bbx_x_min" default="-5" />
  <arg name="bbx_y_max" default="10" />
  <arg name="bbx_y_min" default="-10" />
  <arg name="bbx_z_max" default="5.0" />
  <arg name="bbx_z_min" default="-5.0" />

  <!--turtlebot0 true
      firefly false-->
  <arg name="filter_ground" default="true" />
  <arg name="filter_ground_threshold" default="0.05" />

  <node name="tentabot_map_utility_server" pkg="tentabot" type="tentabot_map_utility_server" output="screen">
  
    <param name="world_frame_name" value="$(arg world_frame_name)" />
    <param name="map_name" value="$(arg map_name)" />
    <param name="map_frame_name" value="$(arg map_frame_name)" />
    <param name="map_resolution" value="$(arg map_resolution)" />
    <param name="map_server_dt" value="$(arg map_server_dt)" />
    <param name="sensor_pc2_msg_name" value="$(arg sensor_pc2_msg_name)" />
    <param name="sensor_pc2_direction" value="$(arg sensor_pc2_direction)" />
    <param name="sensor_pc2_min_range" value="$(arg sensor_pc2_min_range)" />
    <param name="sensor_pc2_max_range" value="$(arg sensor_pc2_max_range)" />
    <param name="sensor_pc2_max_yaw" value="$(arg sensor_pc2_max_yaw)" />
    <param name="sensor_pc2_max_pitch" value="$(arg sensor_pc2_max_pitch)" />
    <param name="sensor_laser_msg_name" value="$(arg sensor_laser_msg_name)" />
    <param name="sensor_laser_max_range" value="$(arg sensor_laser_max_range)" />
    <param name="crop_x_max" value="$(arg crop_x_max)" />
    <param name="crop_x_min" value="$(arg crop_x_min)" />
    <param name="crop_y_max" value="$(arg crop_y_max)" />
    <param name="crop_y_min" value="$(arg crop_y_min)" />
    <param name="crop_z_max" value="$(arg crop_z_max)" />
    <param name="crop_z_min" value="$(arg crop_z_min)" />
    <param name="bbx_x_max" value="$(arg bbx_x_max)" />
    <param name="bbx_x_min" value="$(arg bbx_x_min)" />
    <param name="bbx_y_max" value="$(arg bbx_y_max)" />
    <param name="bbx_y_min" value="$(arg bbx_y_min)" />
    <param name="bbx_z_max" value="$(arg bbx_z_max)" />
    <param name="bbx_z_min" value="$(arg bbx_z_min)" />
    <param name="filter_ground" value="$(arg filter_ground)" />
    <param name="filter_ground_threshold" value="$(arg filter_ground_threshold)" />
  
  </node>

</launch>
